caput/memh5.py

```diff
@@ -2066,28 +2295,28 @@ def get_h5py_File(f, **kwargs):
     # Figure out if F is a file or a filename, and whether the file should be
     # closed.
     if is_group(f):
-        opened = False
-        # if kwargs:
-        #    msg = "Got some keyword arguments but File is alrady open."
-        #    warnings.warn(msg)
+        return f, False
     else:
-        opened = True
+        if file_format is None:
+            file_format = fileformats.guess_file_format(f)
+        if file_format == fileformats.Zarr and not zarr_available:
+            raise RuntimeError("Unable to open zarr file. Please install zarr.")
         try:
-            f = h5py.File(f, **kwargs)
+            f = file_format.open(f, **kwargs)
         except IOError as e:
             msg = "Opening file %s caused an error: " % str(f)
             raise IOError(msg + str(e)) from e
-    return f, opened
+        return f, file_format == fileformats.HDF5
```

Comment:
Why does it not return `True` for `zarr` as well? Are `zarr` files by default "already opened"?

Reply:
`Zarr` fileformats do not necessarily have a `close()` method; we should take this outside of `get_file`, and decide on how that file format gets closed.